packages/plugins/chainflip/src/broker.ts

```diff
@@ -41,15 +41,17 @@ export const assetIdentifierToChainflipTicker = new Map<string, string>([
   ["SOL.USDC-EPJFWDD5AUFQSSQEM2QN1XZYBAPC8G4WEGGKZWYTDT1V", "SolUsdc"],
 ]);
 
-const registerAsBroker = (toolbox: Awaited<ReturnType<typeof ChainflipToolbox>>) => () => {
-  const extrinsic = toolbox.api.tx.swapping?.registerAsBroker?.();
+const registerAsBroker =
+  (toolbox: Awaited<ReturnType<typeof ChainflipToolbox>>) => (address?: string) => {
+    const extrinsic = toolbox.api.tx.swapping?.registerAsBroker?.();
+    const registerAddress = address ?? toolbox.getAddress();
 
-  if (!extrinsic) {
-    throw new SwapKitError("chainflip_broker_register");
-  }
+    if (!(extrinsic && registerAddress)) {
+      throw new SwapKitError("chainflip_broker_register");
+    }
 
-  return toolbox.signAndBroadcast(extrinsic);
-};
+    return toolbox.signAndBroadcast(extrinsic, undefined, registerAddress);
```

Comment:
Modify params to pass object instead of magic ordered values.

Reply:
I will clean up toolbox when I move changes to v4. saw some other things that needs optimizing